keps/sig-node/129447-log-rotate-on-disk-pressure/README.md

```diff
@@ -0,0 +1,451 @@
+              log_size=$(($log_size + 1048576))  # Increment size by 1MB
+              sleep 0.1  # Sleep to control log generation speed
+            done
+            end_time=$(date +%s)
+            echo "Log generation completed in $((end_time - start_time)) seconds"
+      restartPolicy: Never
+  backoffLimit: 4
+```
+File sizes
+```
+-rw-r----- 1 root root  24142862 Jan  1 11:41 0.log
+-rw-r--r-- 1 root root 183335398 Jan  1 11:40 0.log.20250101-113948.gz
+-rw-r--r-- 1 root root 364144934 Jan  1 11:40 0.log.20250101-114003.gz
+-rw-r--r-- 1 root root 487803789 Jan  1 11:40 0.log.20250101-114023.gz
+-rw-r--r-- 1 root root 577188544 Jan  1 11:41 0.log.20250101-114047.gz
+-rw-r----- 1 root root 730449620 Jan  1 11:41 0.log.20250101-114115
+```
+
+### Spec 2
+
+Continuously generating 10Mib with 10 sec sleep in between
+```
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: generate-huge-logs
+spec:
+  template:
+    spec:
+      containers:
+      - name: log-generator
+        image: busybox
+        command: ["/bin/sh", "-c"]
+        args:
+          - |
+            # Generate huge log entries to stdout
+            start_time=$(date +%s)
+            log_size=0
+            target_size=$((4 * 1024 * 1024 * 1024))  # 4 GB target size in bytes
+            while [ $log_size -lt $target_size ]; do
+              # Generate 1 MB of random data and write it to stdout
+              echo "Generating huge log entry at $(date) - $(dd if=/dev/urandom bs=10M count=1 2>/dev/null)"
+              log_size=$(($log_size + 1048576))  # Increment size by 1MB
+              sleep 0.1  # Sleep to control log generation speed
+            done
+            end_time=$(date +%s)
+            echo "Log generation completed in $((end_time - start_time)) seconds"
+      restartPolicy: Never
+  backoffLimit: 4
+```
+
+File sizes
+```
+-rw-r----- 1 root root 181176268 Jan  1 11:31 0.log
+-rw-r--r-- 1 root root 183336647 Jan  1 11:20 0.log.20250101-111730.gz
+-rw-r--r-- 1 root root 183323382 Jan  1 11:23 0.log.20250101-112026.gz
+-rw-r--r-- 1 root root 183327676 Jan  1 11:26 0.log.20250101-112321.gz
+-rw-r--r-- 1 root root 183336376 Jan  1 11:29 0.log.20250101-112616.gz
+-rw-r----- 1 root root 205360966 Jan  1 11:29 0.log.20250101-112911
+```
+
+
+If the pod had been generating logs in Gigabytes with minimal delay, it can cause disk pressure on kubelet host and that can affect other pods running in the same kubelets.
+
+### Goals
+
+- Rotate and Clean all container logs on kubelet Disk pressure
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+### Risks and Mitigations
+
+No identified risk.
```

Comment:
let's think a bit more about this, especially if we start from beta. What can go wrong?